### PR TITLE
Patch eap.o memory leak

### PR DESCRIPTION
**Pull request by mhightower83**
WiFi Enterprise option can leak up to 3 allocations per connect/disconnect cycle (configuration dependent): anonymous Identity, password, and some unidentified allocation.

This solution patches `eap.o` from `libwpa2.a` to call a special 2-part wrapper instead of `vPortFree` at cleanup.

This is supplemental to https://github.com/esp8266/Arduino/pull/8529#issue-1192106505 and should not be merged prior.

There may be a better way of doing this. My knowledge of `xtensa-lx106-elf-objcopy` is near zero and everything that sounded promising didn't work the way I needed. 

I don't have a use case for WiFi Enterprise so my testing has been limited. More testing is needed.

To minimize resources used, the patch will only be linked in when you have a call to `enable_wifi_enterprise_patch()`.

Use something like this to enable patch:
```cpp
#include <ESP8266WiFi.h>
#include <coredecls.h>
// ...
void setup() {
  enable_wifi_enterprise_patch();
// ...
}
```

### CODE REVIEW

File: cores/esp8266/wpa2_eap_patch.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    ".size z2EapFree, .-z2EapFree\n\t"
+);
+
+/*
+ * While some insight can be gained from the ESP32 repo for this structure.
+ * It does not match exactly. This alternate structure focuses on correct offset
+ * rather than trying to exactly reconstruct the original labels.
+ */
+struct StateMachine { // size 200 bytes
+    void* befoeConfig[16];
```
Comment (Flole998) [inline cores/esp8266/wpa2_eap_patch.cpp:70]: Not that it matters, but you probably meant `beforeConfig` (you're missing the r here).

**Comment by Flole998** [comment]
Excellent job, I am really impressed. I tried to do something similar but failed miserably.

File: cores/esp8266/wpa2_eap_patch.cpp (hunk 1 above)
Comment (mhightower83) [reply cores/esp8266/wpa2_eap_patch.cpp:70]: Thanks! I need to get a new keyboard. I think the keys bind a bit and I don't notice, due to my brain's tendency to auto-complete. I think it is a feature of dyslexia where I cannot see things that are missing. :frowning_face:

**Comment by mhightower83** [comment]
@earlephilhower Thanks! I started using this bug to test some code I was working on to catch a double free. I ended out making several more debug enhancements to umm_malloc. These helped reduce the amount of assembly code I was reading.

**Pull request merged**